api4/post.go

```diff
@@ -396,6 +397,54 @@ func getPost(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getPostsByIds(c *Context, w http.ResponseWriter, r *http.Request) {
+	postIDs := model.ArrayFromJSON(r.Body)
+
+	if len(postIDs) == 0 {
+		c.SetInvalidParam("post_ids")
+		return
+	}
+
+	postsList, err := c.App.GetPostsByIds(postIDs)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	var posts = []*model.Post{}
+
+	for _, post := range postsList {
+
+		channel, err := c.App.GetChannel(post.ChannelId)
+		if err != nil {
+			c.Err = err
+			return
+		}
+
+		if !c.App.SessionHasPermissionToChannel(*c.AppContext.Session(), channel.Id, model.PermissionReadChannel) {
+			if channel.Type == model.ChannelTypeOpen {
+				if !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), channel.TeamId, model.PermissionReadPublicChannel) {
```

Comment:
Seems like both these conditions can be combined with an `&&`?

Reply:
I don't think they can be combined with `&&`:

If a channel is set to open and `hasPermissionToTeam` is evaluated to `true`, the post would not be included in the list if they conditions are combined with `&&`

Reply:
Ah my apologies, it's because of the `else` condition. Would something like this work?

```
if !open || (open && !hasPermission()) {
  continue
}
```

0/5.

Reply:
That works, updated